types/src/index.ts

```diff
@@ -384,6 +384,11 @@ export interface Action<
   type: "action";
   methods?: Methods & ThisType<PropThis<ActionPropDefinitions> & Methods>;
   props?: ActionPropDefinitions;
+  annotations?: {
```

Comment:
you should add `idempotentHint?` while you're at it, even though we aren't backfilling that value for any components right now